src/index.ts

```diff
@@ -22,14 +24,23 @@ export interface SpawnOptions {
   pubsub?: boolean
   pubsubRouter?: PubSubRouter
   muxer?: Muxer
+  relay?: boolean
+  // the node will not listen on any
+  // addresses if true
+  noListen?: boolean
 }
 
 export interface DaemonFactory {
   spawn: (options: SpawnOptions) => Promise<Daemon>
+  relay?: {
+    // make a reservation on a relay
+    reserve: (daemon: Daemon, peerId: PeerId) => Promise<void>
```

Comment:
Creating a reservation on a relay could be handled in the following ways:
1. implementing it in the daemons themselves and exposing functionality through the daemon protocol.
2. Using existing stream handlers on the daemon to create a reservation.
If we pass in a function to handle stream reservation, we can use the existing proto definitions for circuitv2 in `js-libp2p`.